### PR TITLE
documentation refactor

### PR DESCRIPTION
**Pull request by leqiao-1**
## Describe your changes
Documentation update

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`

### CODE REVIEW

File: docs/source/overview/quicktour.md [hunk 1]
```diff
@@ -223,9 +119,25 @@ python -m olive.workflows.run --config config.json
...
+It is composed with 5 parts:
+### [Input Model](../tutorials/options.md/#input-model-information)
+You provide input model location and type. PyTorchModel, ONNXModel, OpenVINOModel and SNPEModel are supported model types.
+
+### [Host and Target Systems](../tutorials/options.md/#systems-information)
+An optimization technique, which we call a Pass, can be run on a variety of host systems and the resulting model evaluated on desired target systems. More details for the available systems can be found at at [OliveSystems api reference](systems).
+
+### [Evaluator](../tutorials/options.md/#evaluators-information)
+In order to chose the set of Pass configuration parameters that lead to the “best” model, Olive requires an evaluator that returns metrics values for each output model.
+
```
Comment (devang-ml) [inline docs/source/overview/quicktour.md:131]: You list your performance requirements (for example, accuracy, latency), that optimized candidate models should meet, here.

File: docs/source/index.rst [hunk 2]
```diff
@@ -23,10 +21,18 @@ This document introduces Olive and provides some examples to get you started.
...
+
+   examples
+
 .. toctree::
    :maxdepth: 1
    :caption: TUTORIALS
 
+   tutorials/design
+   tutorials/options
    tutorials/configure_systems
```
Comment (devang-ml) [inline docs/source/index.rst:36]: List of options should be part of the user guide, not tutorials.

File: docs/source/overview/quicktour.md [hunk 3]
```diff
@@ -304,3 +216,5 @@ from olive.engine.footprint import Footprint
 footprint = Footprint.from_file("footprints.json")
 footprint.plot_pareto_frontier()
 ```
+### Olive Packaging
+Olive also can package output artifacts when user adds `PackagingConfig` to Engine configurations. Please refer to [Packaing Olive artifacts](../tutorials/packaging_output_models.md) for more details.
```
Comment (devang-ml) [inline docs/source/overview/quicktour.md:220]: We should include packaging config in this quick tour example.

File: docs/source/index.rst [hunk 4]
```diff
@@ -23,10 +21,18 @@ This document introduces Olive and provides some examples to get you started.
...
+   :caption: EXAMPLES
+
+   examples
+
 .. toctree::
    :maxdepth: 1
    :caption: TUTORIALS
 
+   tutorials/design
+   tutorials/options
```
Comment (devang-ml) [inline docs/source/index.rst:35]: There are two kinds of tutorials and we should separate them in subsections
- Tutorials for Model developers (who are Olive users)
- Tutorials for Olive developers (who are Olive contributors)

File: docs/source/overview/quicktour.md (hunk 3 above)
Comment (leqiao-1) [reply docs/source/overview/quicktour.md]: included a config exmaple and a result folder structure example

File: docs/source/index.rst (hunk 2 above)
Comment (leqiao-1) [reply docs/source/index.rst:36]: Move options back to overview

File: docs/source/index.rst (hunk 4 above)
Comment (leqiao-1) [reply docs/source/index.rst]: Currently only "How to add new Pass" is tutorial for contributors. I move this part to "Guide for contributors"

File: docs/source/overview/quicktour.md (hunk 1 above)
Comment (leqiao-1) [reply docs/source/overview/quicktour.md:131]: added

**Review by devang-ml — APPROVED** [review]

File: docs/source/examples.md [hunk 5]
```diff
@@ -1,26 +1,20 @@
...
-[directml/stable_diffusion](https://github.com/microsoft/Olive/tree/main/examples/directml/stable_diffusion)
-
-# BERT optimization with Intel® Neural Compressor Post Training quantization on CPU
-[bert_inc_ptq_cpu](https://github.com/microsoft/Olive/tree/main/examples/bert_inc_ptq_cpu)
-
-# Whisper optimization using ORT toolchain
-[whisper](https://github.com/microsoft/Olive/tree/main/examples/whisper)
+[Stable Diffusion optimization](https://github.com/microsoft/Olive/tree/main/examples/directml/stable_diffusion)
+# Whisper Acceleration Toolchains
+[Whisper Optimization](https://github.com/microsoft/Olive/tree/main/examples/whisper)
```
[the remark this answers is not in the document]
Comment (leqiao-1) [reply docs/source/examples.md]: Moved into CPU Acceleration

File: docs/source/examples.md [hunk 6]
```diff
@@ -1,26 +1,20 @@
...
-# ResNet optimization with QAT Default Training Loop on CPU
-[resnet_qat_default_train_loop_cpu](https://github.com/microsoft/Olive/tree/main/examples/quantization_aware_training/resnet_qat_default_train_loop_cpu)
+[ResNet optimization with QAT PyTorch Lightning Module](https://github.com/microsoft/Olive/tree/main/examples/quantization_aware_training/resnet_qat_lightning_module_cpu)
+# Acceleration on NPU
+[Inception model optimization on Qualcomm NPU](https://github.com/microsoft/Olive/tree/main/examples/snpe/inception_snpe_qualcomm_npu)
 
-# ResNet optimization with QAT PyTorch Lightning Module on CPU
-[resnet_qat_lightning_module_cpu](https://github.com/microsoft/Olive/tree/main/examples/quantization_aware_training/resnet_qat_lightning_module_cpu)
+# Acceleration with OpenVINO
+[Cifar10 optimization with OpenVINO for Intel HW](https://github.com/microsoft/Olive/tree/main/examples/cifar10_openvino_intel_hw)
```
[the remark this answers is not in the document]
Comment (leqiao-1) [reply docs/source/examples.md:14]: Moved into CPU Acceleration

File: docs/source/examples.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (leqiao-1) [reply docs/source/examples.md]: Moved into CPU Acceleration

File: docs/source/examples.md [hunk 7]
```diff
@@ -1,26 +1,20 @@
...
 
-# SqueezeNet latency optimization with DirectML
-[directml/squeezenet](https://github.com/microsoft/Olive/tree/main/examples/directml/squeezenet)
+# Acceleration with DirectML
```
[the remark this answers is not in the document]
Comment (leqiao-1) [reply docs/source/examples.md]: changed

**Review by EmmaNingMS — CHANGES_REQUESTED** [review]

File: docs/source/overview/quicktour.md [hunk 8]
```diff
@@ -228,61 +80,47 @@ python -m olive.workflows.run --config config.json
...
-    cmp_direction: Dict = None
-    is_goals_met: bool = False
-```
+### [Evaluator](../overview/options.md/#evaluators-information)
+You specify your performance requirements in evaluator, such as accuracy and latency, which the optimized candidate models should meet. Olive utilizes the information to tune the optimal set of optimization parameters for the "best" model.
+
+### [Passes](../overview/options.md/#passes-information)
+An optimization technique is called as a Pass in Olive. You list optimizations that you want to apply on the input model. In this example, we first convert the pytorch model to ONNX then quantize it.
+
+### [Engine](../overview/options.md/#engine-information)
```
[the remark this answers is not in the document]
Comment (leqiao-1) [reply docs/source/overview/quicktour.md:115]: Engine is not optional. It has to define the evaluator used

**Review by EmmaNingMS — APPROVED** [review]
